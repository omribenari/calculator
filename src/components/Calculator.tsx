import { Container, Grid } from '@mantine/core';
import CalcDisplay from './CalcDisplay';
import CalcButton from './CalcButton';

const Calculator = () => {
  return (
    <Container w={300} mx="auto">
      <Grid gutter={3}>
        <Grid.Col span={12}>
          <CalcDisplay />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="7" />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="8" />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="9" />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="*" />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="4" />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="5" />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="6" />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="%" />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="1" />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="2" />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="3" />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="+" />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="0" />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="." />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="AC" />
        </Grid.Col>
        <Grid.Col span={3}>
          <CalcButton text="=" />
        </Grid.Col>
        <Grid.Col span={12}></Grid.Col>
      </Grid>
    </Container>
  );
};

export default Calculator;
