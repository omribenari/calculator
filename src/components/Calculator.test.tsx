import React from 'react';
import { render } from '@testing-library/react';
import Calculator from './Calculator';
import { MemoryRouter } from 'react-router-dom';

describe('Calculator component', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Calculator />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
