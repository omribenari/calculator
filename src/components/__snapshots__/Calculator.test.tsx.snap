// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mantine-Container-root mantine-1v5109t"
  >
    <div
      class="mantine-Grid-root mantine-1eow0dc"
    >
      <div
        class="mantine-Grid-col mantine-gt3psn"
      >
        <div
          class="mantine-Text-root mantine-1v82c8f"
          title=""
        />
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="7"
        >
          7
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="8"
        >
          8
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="9"
        >
          9
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="*"
        >
          *
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="4"
        >
          4
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="5"
        >
          5
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="6"
        >
          6
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="%"
        >
          %
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="1"
        >
          1
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="2"
        >
          2
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="3"
        >
          3
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="+"
        >
          +
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="0"
        >
          0
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="."
        >
          .
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="AC"
        >
          AC
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-1bp3v81"
      >
        <div
          class="mantine-Text-root mantine-1tvbygw"
          title="="
        >
          =
        </div>
      </div>
      <div
        class="mantine-Grid-col mantine-gt3psn"
      />
    </div>
  </div>
</DocumentFragment>
`;
